### PR TITLE
Update coupling config for ScenarioMIP

### PR DESCRIPTION
**Pull request by laurinks**
## Purpose of this PR

Set `cfg$gms$s15_elastic_demand <- 1` in coupling config for ScenarioMIP, and add it to known column names.

Set` cfg_mag$gms$s32_npi_ndc_reversal` and `cfg_mag$gms$s35_npi_ndc_reversal` to 2030 in H scenarios for NPi and NDC policy reversal.

Set **carbonprice** and **techpol** to `none` for H scenarios.

## Type of change

- [x] Minor change (default scenarios show only small differences)

## Checklist:

- [x] My code follows the [coding etiquette](https://github.com/remindmodel/remind/blob/develop/main.gms#L80)
- [x] I performed a self-review of my own code
- [x] I explained my changes within the PR, particularly in hard-to-understand areas
- [x] I checked that the [in-code documentation](https://github.com/remindmodel/remind/blob/develop/main.gms#L120) is up-to-date
- [ ] I adjusted the reporting in [`remind2`](https://github.com/pik-piam/remind2) where it was needed
- [ ] I adjusted `forbiddenColumnNames` in [readCheckScenarioConfig.R](https://github.com/remindmodel/remind/blob/develop/scripts/start/readCheckScenarioConfig.R) in case the PR leads to deprecated switches
- [x] All automated model tests pass (`FAIL 0` in the output of `make test`)
- [ ] The changelog `CHANGELOG.md` [has been updated correctly](https://gitlab.pik-potsdam.de/rse/rsewiki/-/wikis/Standards-for-Writing-a-Changelog)

### CODE REVIEW

**Comment by laurinks** [comment]
Florian requested this change to the ScenarioMIP coupling config. @flohump , @dklein-pik : Could you have a quick look that is as intended and is set correctly? Thanks!

File: config/scenario_config_coupled_ScenarioMIP.csv [hunk 1]
```diff
@@ -1,40 +1,40 @@
...
-SMIPv04-ML-SSP2-PkBudg2000-var_EWoff;1;SMIPv04-ML-SSP2-PkBudg2000-def;;;;;;;;;
-SMIPv04-L-SSP2-PkBudg1000-var_EWoff;1;SMIPv04-L-SSP2-PkBudg1000-def;;;;;;;;;
-SMIPv04-VLHO-SSP2-EcBudg400-var_EWoff;1;SMIPv04-VLHO-SSP2-EcBudg400-def;;;;;;;;;
-SMIPv04-VLLO-SSP1-PkBudg750-var_EWoff;1;SMIPv04-VLLO-SSP1-PkBudg750-def;;;;;;;;;
-SMIPv04-VLLO-SSP1-PkBudg750-var_CESMkUps;1;SMIPv04-VLLO-SSP1-PkBudg750-def;;;;;;;;;
-SMIPv04-VLLO-SSP1-PkBudg750-var_ProdCost;1;SMIPv04-VLLO-SSP1-PkBudg750-def;;;;;;;;;
-SMIPv04-VLLO-SSP1-PkBudg750-var_bioenergy_SustTax1p75;1;SMIPv04-VLLO-SSP1-PkBudg750-def;;;;;;;;;
-SMIPv04-VLLO-SSP1-PkBudg750-var_tradecostBio0p5;1;SMIPv04-VLLO-SSP1-PkBudg750-def;;;;;;;;;
-SMIPv04-VLLO-SSP1-PkBudg750-var_maxProdBiolc100;1;SMIPv04-VLLO-SSP1-PkBudg750-def;;;;;;;;;
+title;start;copyConfigFrom;oldrun;path_report;qos;config/scenario_config.csv;no_ghgprices_land_until;var_luc;path_gdx;path_gdx_bau;path_gdx_ref;cfg$gms$s15_elastic_demand
```
Comment (dklein-pik) [inline config/scenario_config_coupled_ScenarioMIP.csv:1]:
```suggestion
title;start;copyConfigFrom;oldrun;path_report;qos;config/scenario_config.csv;no_ghgprices_land_until;var_luc;path_gdx;path_gdx_bau;path_gdx_ref;cfg_mag$gms$s15_elastic_demand
```
To be identified by the coupling scripts individual MAgPIE switches need to start with `cfg_mag`.

**Review by dklein-pik — APPROVED** [review]

File: config/scenario_config_coupled_ScenarioMIP.csv [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
+SMIPv04-M-SSP2-NPi2025-var_EWoff;1;SMIPv04-M-SSP2-NPi2025-def;;;;;;;;;;-1
+SMIPv04-ML-SSP2-PkBudg2000-var_EWoff;1;SMIPv04-ML-SSP2-PkBudg2000-def;;;;;;;;;;-1
+SMIPv04-L-SSP2-PkBudg1000-var_EWoff;1;SMIPv04-L-SSP2-PkBudg1000-def;;;;;;;;;;-1
+SMIPv04-VLHO-SSP2-EcBudg400-var_EWoff;1;SMIPv04-VLHO-SSP2-EcBudg400-def;;;;;;;;;;-1
+SMIPv04-VLLO-SSP1-PkBudg750-var_EWoff;1;SMIPv04-VLLO-SSP1-PkBudg750-def;;;;;;;;;;-1
+SMIPv04-VLLO-SSP1-PkBudg750-var_CESMkUps;1;SMIPv04-VLLO-SSP1-PkBudg750-def;;;;;;;;;;-1
+SMIPv04-VLLO-SSP1-PkBudg750-var_ProdCost;1;SMIPv04-VLLO-SSP1-PkBudg750-def;;;;;;;;;;-1
+SMIPv04-VLLO-SSP1-PkBudg750-var_bioenergy_SustTax1p75;1;SMIPv04-VLLO-SSP1-PkBudg750-def;;;;;;;;;;-1
+SMIPv04-VLLO-SSP1-PkBudg750-var_tradecostBio0p5;1;SMIPv04-VLLO-SSP1-PkBudg750-def;;;;;;;;;;-1
+SMIPv04-VLLO-SSP1-PkBudg750-var_maxProdBiolc100;1;SMIPv04-VLLO-SSP1-PkBudg750-def;;;;;;;;;;-1
```
Comment (flohump) [inline config/scenario_config_coupled_ScenarioMIP.csv:40]:
```suggestion
SMIPv04-VLLO-SSP1-PkBudg750-var_maxProdBiolc100;1;SMIPv04-VLLO-SSP1-PkBudg750-def;;;;;;;;;;1
```

Comment (flohump) [reply config/scenario_config_coupled_ScenarioMIP.csv:40]: it should be set to `1` not `-1`

File: scripts/start/readCheckScenarioConfig.R [hunk 3]
```diff
@@ -145,7 +145,7 @@ readCheckScenarioConfig <- function(filename, remindPath = ".", testmode = FALSE
   if (coupling %in% "MAgPIE") {
     knownColumnNames <- c(knownColumnNames, "cm_nash_autoconverge_lastrun", "oldrun", "path_report", "magpie_scen",
                           "no_ghgprices_land_until", "qos", "sbatch", "path_mif_ghgprice_land", "max_iterations",
-                          "magpie_empty", "var_luc")
+                          "magpie_empty", "var_luc","cfg$gms$s15_elastic_demand")
```
Comment (flohump) [inline scripts/start/readCheckScenarioConfig.R:148]: @dklein-pik Does this work as suggested?

**Review by flohump — CHANGES_REQUESTED** [review]
fine in general. But the value should be `1` not `-1`

File: scripts/start/readCheckScenarioConfig.R (hunk 3 above)
Comment (dklein-pik) [reply scripts/start/readCheckScenarioConfig.R:148]: Oh, thanks, Flo, this must of cause also be renamed to `cfg_mag$...`

File: config/scenario_config_coupled_ScenarioMIP.csv (hunk 2 above)
Comment (laurinks) [reply config/scenario_config_coupled_ScenarioMIP.csv]: Sorry, I misread the `<- 1` as a `-1`. Changed now.

**Review by flohump — APPROVED** [review]
All fine now.

**Comment by nicobauer** [comment]
Hi @laurinks , thanks. I am a bit confused now about the policy roll back. Florian sent me these parameters: 
cfg_mag$gms$s32_npi_ndc_reversal and the same with s35. I do not find it in your files.

**Comment by laurinks** [comment]
@flohump : Brief follow-up question from Nico: If we set the `s32/35_npi_ndc_reversal` switches to 2030, this means that in 2030 there is already no more NPi and NDC policy? Or does it mean that the reversal happens after 2030, i.e. no more NPi and NDC policy from 2035? Thanks!

**Comment by flohump** [comment]
> @flohump : Brief follow-up question from Nico: If we set the `s32/35_npi_ndc_reversal` switches to 2030, this means that in 2030 there is already no more NPi and NDC policy? Or does it mean that the reversal happens after 2030, i.e. no more NPi and NDC policy from 2035? Thanks!

Setting `s32/35_npi_ndc_reversal` to 2030 means that from 2030 onward the NPI policy is deactivated. 
But MAgPIE will keep the TC effects gained earlier due to the NPI policy. 
I tested it, there is no abrubt shift towards deforestation.

**Review by flohump — APPROVED** [review]
All fine if 2030 alings with the timing of NPI policy rollback assumed in REMIND

**Comment by laurinks** [comment]
Yes, this should be in line with Nico's plans for REMIND. The settings in the ScenarioMIP config for the H scenario will likely still evolve as @nicobauer is working on it.

**Comment by nicobauer** [comment]
Thanks a lot, Laurin and Florian!

**Pull request merged**